### DISCUSSION TITLE
Map vs Reduce

### DISCUSSION THREAD

**Discussion by Rastraponovich** [nodejs/nodejs.org#6587] [Ideas]
why are you using 2(3) map functions instead of 1 reduce?
[in this](https://github.com/nodejs/nodejs.org/blob/cf31dafbd64e66c2a1b412ead38c8a624c56eaeb/components/Common/BlogPostCard/index.tsx#L34)

**Comment by ovflowd** [comment]
Because, yes. Each data structure is used for a different purpose. The first `map` is used to pass `avatars` to the `AvatarGroups` component whilst the 2nd is to simply get the name of the person. To be honest, the 2nd map could come from `authors` instead of `avatars` but the end effect is the same. Only one reduce wouldn't suffice.

The 2nd map is being used to convert the object structure into a scalar so we can "join" the values of the object. To be honest, we could use a `toString` function within the original object.

In performance terms, this is negligible since these arrays would never pass the dozens of entries. Even with millions of entries this mapping is simple enough.
